cluster-autoscaler/cloudprovider/mcm/mcm_cloud_provider.go

```diff
@@ -38,6 +39,9 @@ import (
 const (
 	// ProviderName is the cloud provider name for MCM
 	ProviderName = "mcm"
+
+	//MachineTypeNotAvailableAnnotation is an annotation put by MCM on machine-deployment if certain machine-types are not available.
+	MachineTypeNotAvailableAnnotation = "machine.sapcloud.io/machine-type-not-available"
```

Comment:
The propagation of this label from machine/machineDeployment controller till the autoscaler seems like a hack. Is there a more elegent solution here? If not can we keep this hack only at the autoscaler for now. By making the autoscaler search for failed machine patterns in the failedMachines field?

Reply:
Yes sure, I will give it a try with changes only at CA, and not touch MCM that definitely be the less complicated solution.

Reply:
We need to re-investigate the overall approach.